File: app/src/main/java/com/hungtran/bankingassistant/ui/maket/MaketFragment.java
package com.hungtran.bankingassistant.ui.maket;

import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.view.View;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.hungtran.bankingassistant.R;
import com.hungtran.bankingassistant.adapters.RateViewPagerAdapter;
import com.hungtran.bankingassistant.util.Constant;
import com.hungtran.bankingassistant.util.base.BaseFragment;

import butterknife.BindView;
import butterknife.ButterKnife;
import info.hoang8f.android.segmented.SegmentedGroup;

public class MaketFragment extends BaseFragment implements SegmentedGroup.OnCheckedChangeListener, ViewPager.OnPageChangeListener{

    @BindView(R.id.segmented)
    SegmentedGroup mSegmentedGroup;

    @BindView(R.id.rateViewPager)
    ViewPager mViewPager;

    @BindView(R.id.btnInterestRate)
    RadioButton btnInterestRate;

    private static MaketFragment instance;
    RateViewPagerAdapter rateViewPagerAdapter;

    public static MaketFragment getInstance() {
        if (instance == null) {
            instance = new MaketFragment();
        }
        return new MaketFragment();
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_maket;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);
        mViewPager.setOffscreenPageLimit(3);
        setupViewPager();
        mSegmentedGroup.setTintColor(Color.parseColor(Constant.PRIMARY_COLOR));
        mSegmentedGroup.setOnCheckedChangeListener(this);
        mSegmentedGroup.check(R.id.btnInterestRate);
    }

    private void setupViewPager() {
        rateViewPagerAdapter = new RateViewPagerAdapter(getFragmentManager());
        mViewPager.setAdapter(rateViewPagerAdapter);
        mViewPager.addOnPageChangeListener(this);
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        switch (checkedId) {
            case R.id.btnInterestRate:
                mViewPager.setCurrentItem(0);
                break;
            case R.id.btnExchangeRate:
                mViewPager.setCurrentItem(1);
                break;
            case R.id.btnGold:
                mViewPager.setCurrentItem(2);
                break;
        }
    }

    @Override
    public void onPageScrolled(int i, float v, int i1) {

    }

    @Override
    public void onPageSelected(int i) {
       switch (i) {
           case 0: mSegmentedGroup.check(R.id.btnInterestRate); break;
           case 1: mSegmentedGroup.check(R.id.btnExchangeRate); break;
           case 2: mSegmentedGroup.check(R.id.btnGold); break;
       }
    }

    @Override
    public void onPageScrollStateChanged(int i) {

    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/hungtran/bankingassistant/ui/maket/MaketFragment.kt
package com.hungtran.bankingassistant.ui.maket

import android.graphics.Color
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.view.View
import android.widget.RadioGroup
import com.hungtran.bankingassistant.R
import com.hungtran.bankingassistant.adapters.RateViewPagerAdapter
import com.hungtran.bankingassistant.util.base.BaseFragment
import kotlinx.android.synthetic.main.fragment_rate.*

/**
 * Created by hungtd on 2/18/19.
 */

class MaketFragment : BaseFragment(), ViewPager.OnPageChangeListener, RadioGroup.OnCheckedChangeListener {

    override fun getLayoutId(): Int {
        return R.layout.fragment_rate
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        segmented.setTintColor(Color.parseColor("#008577"))
        segmented.setOnCheckedChangeListener(this)
        btnInterestRate.isChecked = true
        val rateViewPagerAdapter = RateViewPagerAdapter(fragmentManager!!)
        rateViewPager.adapter = rateViewPagerAdapter
        rateViewPager.addOnPageChangeListener(this)
    }

    override fun onPageScrollStateChanged(p0: Int) {

    }

    override fun onPageScrolled(p0: Int, p1: Float, p2: Int) {

    }

    override fun onPageSelected(p0: Int) {
      when(p0) {
          0 -> btnInterestRate.isChecked = true
          1 -> btnExchangeRate.isChecked = true
          2 -> btnGold.isChecked = true
      }
    }

    override fun onCheckedChanged(group: RadioGroup?, checkedId: Int) {
        when(checkedId) {
            R.id.btnInterestRate -> rateViewPager.setCurrentItem(0,true)
            R.id.btnExchangeRate -> rateViewPager.setCurrentItem(1,true)
            R.id.btnGold -> rateViewPager.setCurrentItem(2,true)
        }
    }

}
